cmd/olm/main.go

```diff
@@ -128,98 +128,198 @@ func main() {
 	http.Handle("/metrics", promhttp.Handler())
 	go http.ListenAndServe(":8081", nil)
 
-	ready, done := operator.Run(stopCh)
+	ready, done, sync := operator.Run(stopCh)
 	<-ready
 
 	if *writeStatusName != "" {
-		opStatusGV := schema.GroupVersion{
-			Group:   "config.openshift.io",
-			Version: "v1",
+		monitorClusterStatus(sync, stopCh, opClient, configClient)
+	}
+
+	<-done
+}
+
+func monitorClusterStatus(syncCh chan error, stopCh <-chan struct{}, opClient operatorclient.ClientInterface, configClient configv1client.ConfigV1Interface) {
+	var (
+		syncs              int
+		successfulSyncs    int
+		hasClusterOperator bool
+	)
+	go wait.Until(func() {
+		// slow poll until we see a cluster operator API, which could be never
+		if !hasClusterOperator {
+			opStatusGV := schema.GroupVersion{
+				Group:   "config.openshift.io",
+				Version: "v1",
+			}
+			err := discovery.ServerSupportsVersion(opClient.KubernetesInterface().Discovery(), opStatusGV)
+			if err != nil {
+				log.Infof("ClusterOperator api not present, skipping update (%v)", err)
+				time.Sleep(time.Minute)
+				return
+			}
+			hasClusterOperator = true
 		}
-		err := discovery.ServerSupportsVersion(opClient.KubernetesInterface().Discovery(), opStatusGV)
-		if err != nil {
-			log.Infof("ClusterOperator api not present, skipping update (%v)", err)
-		} else {
-			existing, err := configClient.ClusterOperators().Get(*writeStatusName, metav1.GetOptions{})
-			if k8serrors.IsNotFound(err) {
-				log.Info("Existing operator status not found, creating")
-				created, err := configClient.ClusterOperators().Create(&configv1.ClusterOperator{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: *writeStatusName,
-					},
-				})
-				if err != nil {
-					log.Fatalf("ClusterOperator create failed: %v\n", err)
+
+		// Sample the sync channel and see whether we're successfully retiring syncs as a
```

Comment:
Did you mean "retrying", not "retiring"?

Reply:
No, retiring.  Term for when you complete something and don't have to deal with it anymore - you "retire a transaction from a retry loop"